src/components/StateSelector.tsx

```diff
@@ -89,7 +89,8 @@ function StateSelector(
             brickRoadIndicator={errorText ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : undefined}
             errorText={errorText}
             onPress={() => {
-                const activeRoute = Navigation.getActiveRoute();
+                // Check the route has policyID because getActiveRouteWithoutParams is used in profile page only
+                const activeRoute = /[?&]policyID=([^&]+)/.test(Navigation.getActiveRoute()) ? Navigation.getActiveRoute() : Navigation.getActiveRouteWithoutParams();
```

Comment:
Let's not do that. The root cause is still unclear.

> If we look at the param in the URL path when selecting state, we'll see that &country=AQ (old country code) is there

Why we have the country param in the first place? It was not on the original url

Reply:
I will find another solution

Reply:
I'll update today

Reply:
> Why we have the country param in the first place? It was not on the original URL

in [here](https://github.com/Expensify/App/blob/28a9a37f964f94e205729165fad7c8c571e54dba/src/components/CountrySelector.tsx#L63) we set `country = undefined` to ensure the component country is driven by the parent component rather than URL parameters.

When we choose Adress line 1 we should update country params and path [here](https://github.com/Expensify/App/blob/28a9a37f964f94e205729165fad7c8c571e54dba/src/components/AddressSearch/index.tsx#L246)

```
      Navigation.setParams({
            country,
        })
      Navigation.navigate(`${activeRoute}&country=${country}` as Route)
```

@s77rt  what do you think?

Reply:
We already have another PR fixing the bug https://github.com/Expensify/App/pull/45394